Apps/Examples/Examples/All Examples/DebugMapExample.swift

```diff
@@ -0,0 +1,179 @@
+            target: self,
+            action: #selector(saveSettings(_:)))
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        preferredContentSize = listView.contentSize
+    }
+
+    @objc private func saveSettings(_ sender: UIBarButtonItem) {
+        delegate?.debugOptionSettingsDidChange(self)
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        allSettings.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cellID = String(describing: DebugOptionCell.self)
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellID) as? DebugOptionCell ?? DebugOptionCell()
+
+        let setting = allSettings[indexPath.row]
+        cell.configure(with: setting, isOptionEnabled: enabledDebugOptions.contains(setting.debugOption))
+        cell.onToggled { [unowned self] isEnabled in
+            if !isEnabled {
+                self.enabledDebugOptions.removeAll(where: { $0 == setting.debugOption })
+            } else if !self.enabledDebugOptions.contains(setting.debugOption) {
+                self.enabledDebugOptions.append(setting.debugOption)
+            }
+        }
+
+        return cell
+    }
+}
+
+// MARK: Cell
+
+private class DebugOptionCell: UITableViewCell {
+
+    private let titleLabel = UILabel()
+    private let toggle = UISwitch()
+
+    init() {
+        super.init(style: .default, reuseIdentifier: String(describing: Self.self))
+        toggle.addTarget(self, action: #selector(didToggle(_:)), for: .valueChanged)
+
+        contentView.addSubview(titleLabel)
+        contentView.addSubview(toggle)
+
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        toggle.translatesAutoresizingMaskIntoConstraints = false
+
+        let constraints: [NSLayoutConstraint] = [
+            titleLabel.leftAnchor.constraint(equalTo: contentView.leftAnchor, constant: 16),
+            titleLabel.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
+            titleLabel.topAnchor.constraint(greaterThanOrEqualTo: contentView.topAnchor, constant: 8),
+            toggle.leftAnchor.constraint(greaterThanOrEqualTo: titleLabel.rightAnchor, constant: 16),
+            toggle.rightAnchor.constraint(equalTo: contentView.rightAnchor, constant: -16),
+            toggle.centerYAnchor.constraint(equalTo: titleLabel.centerYAnchor),
+            toggle.topAnchor.constraint(greaterThanOrEqualTo: contentView.topAnchor, constant: 8),
+        ]
+        NSLayoutConstraint.activate(constraints)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(with setting: MapDebugOptionSetting, isOptionEnabled: Bool) {
+        titleLabel.text = setting.displayTitle
+        toggle.isOn = isOptionEnabled
+    }
+
+    private var _onToggled: ((Bool) -> Void)?
```

Comment:
Maybe a different name can be used to avoid underscoring the variable? E.g. `onToggledHandler`. Underscored properties are uncharacteristic for Swift code.

Reply:
I think using name with underscore prefix is fine in Swift, but if you can send me some materials to read more about this it would be great. I do like your suggested name better :-D I made this change, thank you.

Reply:
The API Design Guidelines covers this quite well, though it is quite vague sometimes.

I think this convention from the [guidelines](https://www.swift.org/documentation/api-design-guidelines/#general-conventions) applies here:
> Follow case conventions. Names of types and protocols are UpperCamelCase. Everything else is lowerCamelCase.